drivers/segger/Make.defs

```diff
@@ -27,11 +27,11 @@ ifeq ($(CONFIG_SEGGER_RTT),y)
   CFLAGS += -Wno-shadow -Wno-array-bounds
 
   ifneq ($(CONFIG_SEGGER_RTT_SECTION),"")
-    CFLAGS += ${shell $(DEFINE) "$(CC)" SEGGER_RTT_SECTION=CONFIG_SEGGER_RTT_SECTION}
+    ${call DEFINE,CFLAGS,SEGGER_RTT_SECTION=CONFIG_SEGGER_RTT_SECTION}
```

Comment:
let's DEFINE return value like this?
CFLAGS += $(call DEFINE, SEGGER_RTT_SECTION=CONFIG_SEGGER_RTT_SECTION)

Reply:
The custom function of the makefile cannot return a value, it can only execute some commands, so this method cannot be used

I'm using variable concatenation now and it seems a bit better

https://stackoverflow.com/questions/29091370/how-can-i-pass-and-return-a-value-from-user-defined-function-in-makefile